tests/tests_fabric/test_wrappers.py

```diff
@@ -104,6 +104,35 @@ def __init__(self, module):
     warning_cache.clear()
 
 
+def test_fabric_module_setattr():
```

Comment:
Could we extend this test case by adding 
1. A call that sets a submodule (simulating compliance with setting modules onto modules in PyTorch)
2. Defining a property-setter on the original module

These are some edge-cases that come to mind, so maybe good to assert they are working as expected